diff --git a/node_modules/expo-dynamic-app-icon/README.md b/node_modules/expo-dynamic-app-icon/README.md
index 96b4770..f0d48f6 100644
--- a/node_modules/expo-dynamic-app-icon/README.md
+++ b/node_modules/expo-dynamic-app-icon/README.md
@@ -19,11 +19,13 @@ add plugins in `app.json`
         {
           "red": { // icon name
             "image": "./assets/icon1.png", // icon path
-            "prerendered": true // for ios UIPrerenderedIcon option
+            "prerendered": true, // for ios UIPrerenderedIcon option
+            "platforms":  ["ios", "android"]  // optional platforms array. defaults to both platforms if emitted
           },
           "gray": {
             "image": "./assets/icon2.png",
-            "prerendered": true
+            "prerendered": true,
+            "platforms":  ["ios"]
           }
         }
       ]
diff --git a/node_modules/expo-dynamic-app-icon/android/build.gradle b/node_modules/expo-dynamic-app-icon/android/build.gradle
index a50832e..6566c7c 100644
--- a/node_modules/expo-dynamic-app-icon/android/build.gradle
+++ b/node_modules/expo-dynamic-app-icon/android/build.gradle
@@ -22,7 +22,7 @@ buildscript {
     if (ext.has("kotlinVersion")) {
       ext.kotlinVersion()
     } else {
-      ext.safeExtGet("kotlinVersion", "1.6.10")
+      ext.safeExtGet("kotlinVersion", "1.8.10")
     }
   }
 
@@ -52,20 +52,23 @@ afterEvaluate {
 
 android {
   namespace = "expo.modules.dynamicappicon"
-  compileSdkVersion safeExtGet("compileSdkVersion", 31)
+  compileSdkVersion safeExtGet("compileSdkVersion", 33)
 
-  compileOptions {
-    sourceCompatibility JavaVersion.VERSION_11
-    targetCompatibility JavaVersion.VERSION_11
-  }
+  def agpVersion = com.android.Version.ANDROID_GRADLE_PLUGIN_VERSION
+  if (agpVersion.tokenize('.')[0].toInteger() < 8) {
+    compileOptions {
+      sourceCompatibility JavaVersion.VERSION_11
+      targetCompatibility JavaVersion.VERSION_11
+    }
 
-  kotlinOptions {
-    jvmTarget = JavaVersion.VERSION_11.majorVersion
+    kotlinOptions {
+      jvmTarget = JavaVersion.VERSION_11.majorVersion
+    }
   }
 
   defaultConfig {
     minSdkVersion safeExtGet("minSdkVersion", 21)
-    targetSdkVersion safeExtGet("targetSdkVersion", 31)
+    targetSdkVersion safeExtGet("targetSdkVersion", 34)
     versionCode 1
     versionName "0.1.0"
   }
diff --git a/node_modules/expo-dynamic-app-icon/plugin/build/withDynamicIcon.d.ts b/node_modules/expo-dynamic-app-icon/plugin/build/withDynamicIcon.d.ts
index 0de104b..8b0c760 100644
--- a/node_modules/expo-dynamic-app-icon/plugin/build/withDynamicIcon.d.ts
+++ b/node_modules/expo-dynamic-app-icon/plugin/build/withDynamicIcon.d.ts
@@ -1,7 +1,10 @@
 import { ConfigPlugin } from "@expo/config-plugins";
-type IconSet = Record<string, {
+type Platform = "ios" | "android";
+type Icon = {
     image: string;
     prerendered?: boolean;
-}>;
+    platforms?: Platform[];
+};
+type IconSet = Record<string, Icon>;
 declare const withDynamicIcon: ConfigPlugin<string[] | IconSet | void>;
 export default withDynamicIcon;
diff --git a/node_modules/expo-dynamic-app-icon/plugin/build/withDynamicIcon.js b/node_modules/expo-dynamic-app-icon/plugin/build/withDynamicIcon.js
index a416495..2de6e96 100644
--- a/node_modules/expo-dynamic-app-icon/plugin/build/withDynamicIcon.js
+++ b/node_modules/expo-dynamic-app-icon/plugin/build/withDynamicIcon.js
@@ -9,7 +9,7 @@ const fs_1 = __importDefault(require("fs"));
 const path_1 = __importDefault(require("path"));
 // @ts-ignore
 const pbxFile_1 = __importDefault(require("xcode/lib/pbxFile"));
-const { addMetaDataItemToMainApplication, getMainApplicationOrThrow, addUsesLibraryItemToMainApplication, } = config_plugins_1.AndroidConfig.Manifest;
+const { getMainApplicationOrThrow, getMainActivityOrThrow, } = config_plugins_1.AndroidConfig.Manifest;
 const androidFolderPath = ["app", "src", "main", "res"];
 const androidFolderNames = [
     "mipmap-hdpi",
@@ -21,10 +21,23 @@ const androidFolderNames = [
 const androidSize = [162, 108, 216, 324, 432];
 const iosFolderName = "DynamicAppIcons";
 const iosSize = 60;
-const iosScales = [2, 3];
+const ipad152Scale = 2.53;
+const ipad167Scale = 2.78;
+const iosScales = [2, 3, ipad152Scale, ipad167Scale];
 function arrayToImages(images) {
     return images.reduce((prev, curr, i) => ({ ...prev, [i]: { image: curr } }), {});
 }
+const findIconsForPlatform = (icons, platform) => {
+    return Object.keys(icons)
+        .filter(key => {
+        const icon = icons[key];
+        if (icon.platforms) {
+            return icon['platforms'].includes(platform);
+        }
+        return true;
+    })
+        .reduce((prev, curr) => ({ ...prev, [curr]: icons[curr] }), {});
+};
 const withDynamicIcon = (config, props = {}) => {
     const _props = props || {};
     let prepped = {};
@@ -34,19 +47,26 @@ const withDynamicIcon = (config, props = {}) => {
     else if (_props) {
         prepped = _props;
     }
-    // for ios
-    config = withIconXcodeProject(config, { icons: prepped });
-    config = withIconInfoPlist(config, { icons: prepped });
-    config = withIconIosImages(config, { icons: prepped });
-    // for aos
-    config = withIconAndroidManifest(config, { icons: prepped });
-    config = withIconAndroidImages(config, { icons: prepped });
+    const iOSIcons = findIconsForPlatform(prepped, "ios");
+    const iOSIconsLength = Object.keys(iOSIcons).length;
+    if (iOSIconsLength > 0) {
+        config = withIconXcodeProject(config, { icons: iOSIcons });
+        config = withIconInfoPlist(config, { icons: iOSIcons });
+        config = withIconIosImages(config, { icons: iOSIcons });
+    }
+    const androidIcons = findIconsForPlatform(prepped, "android");
+    const androidIconsLength = Object.keys(androidIcons).length;
+    if (androidIconsLength > 0) {
+        config = withIconAndroidManifest(config, { icons: androidIcons });
+        config = withIconAndroidImages(config, { icons: androidIcons });
+    }
     return config;
 };
 // for aos
 const withIconAndroidManifest = (config, { icons }) => {
     return (0, config_plugins_1.withAndroidManifest)(config, (config) => {
         const mainApplication = getMainApplicationOrThrow(config.modResults);
+        const mainActivity = getMainActivityOrThrow(config.modResults);
         const iconNamePrefix = `${config.android.package}.MainActivity`;
         const iconNames = Object.keys(icons);
         function addIconActivityAlias(config) {
@@ -60,14 +80,14 @@ const withIconAndroidManifest = (config, { icons }) => {
                         "android:icon": `@mipmap/${iconName}`,
                         "android:targetActivity": ".MainActivity",
                     },
-                    "intent-filter": [
-                        {
-                            action: [{ $: { "android:name": "android.intent.action.MAIN" } }],
-                            category: [
-                                { $: { "android:name": "android.intent.category.LAUNCHER" } },
-                            ],
-                        },
-                    ],
+                    "intent-filter": [...mainActivity["intent-filter"] || [
+                            {
+                                action: [{ $: { "android:name": "android.intent.action.MAIN" } }],
+                                category: [
+                                    { $: { "android:name": "android.intent.category.LAUNCHER" } },
+                                ],
+                            },
+                        ]]
                 })),
             ];
         }
@@ -134,8 +154,14 @@ const withIconAndroidImages = (config, { icons }) => {
 };
 // for ios
 function getIconName(name, size, scale) {
-    const fileName = `${name}-Icon-${size}x${size}`;
+    const fileName = `${name}-Icon-$${size}x${size}`;
     if (scale != null) {
+        if (scale == ipad152Scale) {
+            return `${fileName}@2x~ipad.png`;
+        }
+        if (scale == ipad167Scale) {
+            return `${fileName}@3x~ipad.png`;
+        }
         return `${fileName}@${scale}x.png`;
     }
     return fileName;
@@ -215,9 +241,7 @@ const withIconInfoPlist = (config, { icons }) => {
                 !config.modResults[key]) {
                 config.modResults[key] = {};
             }
-            // @ts-expect-error
             config.modResults[key].CFBundleAlternateIcons = altIcons;
-            // @ts-expect-error
             config.modResults[key].CFBundlePrimaryIcon = {
                 CFBundleIconFiles: ["AppIcon"],
             };
@@ -253,7 +277,7 @@ async function createIconsAsync(config, { icons }) {
             const iconFileName = getIconName(key, iosSize, scale);
             const fileName = path_1.default.join(iosFolderName, iconFileName);
             const outputPath = path_1.default.join(iosRoot, fileName);
-            const scaledSize = scale * iosSize;
+            const scaledSize = Math.ceil(scale * iosSize);
             const { source } = await (0, image_utils_1.generateImageAsync)({
                 projectRoot: config.modRequest.projectRoot,
                 cacheType: "react-native-dynamic-app-icon",
