diff --git a/node_modules/react-native/Libraries/Text/BaseText/RCTBaseTextShadowView.m b/node_modules/react-native/Libraries/Text/BaseText/RCTBaseTextShadowView.m
index c8fd3b1..5344d16 100644
--- a/node_modules/react-native/Libraries/Text/BaseText/RCTBaseTextShadowView.m
+++ b/node_modules/react-native/Libraries/Text/BaseText/RCTBaseTextShadowView.m
@@ -96,8 +96,8 @@ - (NSAttributedString *)attributedTextWithBaseTextAttributes:(nullable RCTTextAt
       RCTRawTextShadowView *rawTextShadowView = (RCTRawTextShadowView *)shadowView;
       NSString *text = rawTextShadowView.text;
       if (text) {
-        NSAttributedString *rawTextAttributedString =
-            [[NSAttributedString alloc] initWithString:[textAttributes applyTextAttributesToText:text]
+        NSMutableAttributedString *rawTextAttributedString =
+            [[NSMutableAttributedString alloc] initWithString:[textAttributes applyTextAttributesToText:text]
                                             attributes:textAttributes.effectiveTextAttributes];
         [attributedText appendAttributedString:rawTextAttributedString];
       }
diff --git a/node_modules/react-native/Libraries/Text/BaseText/RCTBaseTextViewManager.m b/node_modules/react-native/Libraries/Text/BaseText/RCTBaseTextViewManager.m
index ece6876..610ab8f 100644
--- a/node_modules/react-native/Libraries/Text/BaseText/RCTBaseTextViewManager.m
+++ b/node_modules/react-native/Libraries/Text/BaseText/RCTBaseTextViewManager.m
@@ -55,5 +55,6 @@ - (RCTShadowView *)shadowView
 // Special
 RCT_REMAP_SHADOW_PROPERTY(isHighlighted, textAttributes.isHighlighted, BOOL)
 RCT_REMAP_SHADOW_PROPERTY(textTransform, textAttributes.textTransform, RCTTextTransform)
+RCT_REMAP_SHADOW_PROPERTY(url, textAttributes.url, NSString)

 @end
diff --git a/node_modules/react-native/Libraries/Text/RCTTextAttributes.h b/node_modules/react-native/Libraries/Text/RCTTextAttributes.h
index 22fb646..6dd93f0 100644
--- a/node_modules/react-native/Libraries/Text/RCTTextAttributes.h
+++ b/node_modules/react-native/Libraries/Text/RCTTextAttributes.h
@@ -57,6 +57,7 @@ extern NSString *const RCTTextAttributesTagAttributeName;
 @property (nonatomic, strong, nullable) NSNumber *tag;
 @property (nonatomic, assign) UIUserInterfaceLayoutDirection layoutDirection;
 @property (nonatomic, assign) RCTTextTransform textTransform;
+@property (nonatomic, assign, nullable) NSString *url;

 #pragma mark - Inheritance

diff --git a/node_modules/react-native/Libraries/Text/RCTTextAttributes.m b/node_modules/react-native/Libraries/Text/RCTTextAttributes.m
index c832338..e04bd4b 100644
--- a/node_modules/react-native/Libraries/Text/RCTTextAttributes.m
+++ b/node_modules/react-native/Libraries/Text/RCTTextAttributes.m
@@ -95,6 +95,7 @@ - (void)applyTextAttributes:(RCTTextAttributes *)textAttributes
       : _layoutDirection;
   _textTransform =
       textAttributes->_textTransform != RCTTextTransformUndefined ? textAttributes->_textTransform : _textTransform;
+  _url = textAttributes->_url ?: _url;
 }

 - (NSParagraphStyle *)effectiveParagraphStyle
@@ -211,6 +212,10 @@ - (NSParagraphStyle *)effectiveParagraphStyle
     attributes[RCTTextAttributesTagAttributeName] = _tag;
   }

+  if (_url) {
+    attributes[NSLinkAttributeName] = _url;
+  }
+
   return [attributes copy];
 }

diff --git a/node_modules/react-native/Libraries/Text/TextInput/RCTBackedTextInputDelegate.h b/node_modules/react-native/Libraries/Text/TextInput/RCTBackedTextInputDelegate.h
index 7187177..4423249 100644
--- a/node_modules/react-native/Libraries/Text/TextInput/RCTBackedTextInputDelegate.h
+++ b/node_modules/react-native/Libraries/Text/TextInput/RCTBackedTextInputDelegate.h
@@ -40,6 +40,7 @@ NS_ASSUME_NONNULL_BEGIN
 @optional

 - (void)scrollViewDidScroll:(UIScrollView *)scrollView;
+- (BOOL)textView:(UITextView *)textView shouldInteractWithURL:(NSURL *)URL inRange:(NSRange)characterRange interaction:(UITextItemInteraction)interaction;

 @end

diff --git a/node_modules/react-native/Libraries/Text/TextInput/RCTBackedTextInputDelegateAdapter.m b/node_modules/react-native/Libraries/Text/TextInput/RCTBackedTextInputDelegateAdapter.m
index 9dca6a5..75f04c7 100644
--- a/node_modules/react-native/Libraries/Text/TextInput/RCTBackedTextInputDelegateAdapter.m
+++ b/node_modules/react-native/Libraries/Text/TextInput/RCTBackedTextInputDelegateAdapter.m
@@ -254,6 +254,14 @@ - (BOOL)textView:(__unused UITextView *)textView shouldChangeTextInRange:(NSRang
   return NO;
 }

+- (BOOL)textView:(UITextView *)textView shouldInteractWithURL:(NSURL *)URL inRange:(NSRange)characterRange interaction:(UITextItemInteraction)interaction
+{
+  if ([_backedTextInputView.textInputDelegate respondsToSelector:@selector(textView:shouldInteractWithURL:inRange:interaction:)]) {
+    return [_backedTextInputView.textInputDelegate textView:textView shouldInteractWithURL:URL inRange:characterRange interaction:interaction];
+  }
+  return NO;
+}
+
 - (void)textViewDidChange:(__unused UITextView *)textView
 {
   if (_ignoreNextTextInputCall && [_lastStringStateWasUpdatedWith isEqual:_backedTextInputView.attributedText]) {
diff --git a/node_modules/react-native/Libraries/Text/TextNativeComponent.js b/node_modules/react-native/Libraries/Text/TextNativeComponent.js
index 0d59904..e5bfa8f 100644
--- a/node_modules/react-native/Libraries/Text/TextNativeComponent.js
+++ b/node_modules/react-native/Libraries/Text/TextNativeComponent.js
@@ -64,6 +64,7 @@ const virtualTextViewConfig = {
     isHighlighted: true,
     isPressable: true,
     maxFontSizeMultiplier: true,
+    url: true,
   },
   uiViewClassName: 'RCTVirtualText',
 };
diff --git a/node_modules/react-native/Libraries/Text/VirtualText/RCTVirtualTextShadowView.h b/node_modules/react-native/Libraries/Text/VirtualText/RCTVirtualTextShadowView.h
index eed1726..df3ae19 100644
--- a/node_modules/react-native/Libraries/Text/VirtualText/RCTVirtualTextShadowView.h
+++ b/node_modules/react-native/Libraries/Text/VirtualText/RCTVirtualTextShadowView.h
@@ -9,4 +9,6 @@

 @interface RCTVirtualTextShadowView : RCTBaseTextShadowView

+@property (nonatomic, assign, nullable) NSString *url;
+
 @end
